src/python/pants/core/goals/test.py

```diff
@@ -565,6 +565,36 @@ class EnvironmentAware:
         advanced=True,
         help="The maximum timeout (in seconds) that may be used on a test target.",
     )
+    batch_size = IntOption(
+        "--batch-size",
+        default=128,
+        advanced=True,
+        help=softwrap(
+            """
+            The target maximum number of files to be included in each run of batch-enabled
+            test runners.
+
+            Some test runners can execute tests from multiple files in a single run. Plugins
+            that implement this behavior will return all tests that _can_ run together as a
+            single group - the core `test` goal will then divide that group into smaller batches
+            according to this parameter. This is done:
```

Comment:
Put this in terms of users, rather than plugin authors

```suggestion
            Some test runners can execute tests from multiple files in a single run. Test
            implementations will return all tests that _can_ run together as a
            single group - and then this may be further divided into smaller batches,
            based on this option. This is done:
```